File: src/main/java/me/miki/shindo/management/addons/settings/impl/CategorySetting.java
package me.miki.shindo.management.addons.settings.impl;

import me.miki.shindo.Shindo;
import me.miki.shindo.management.addons.Addon;
import me.miki.shindo.management.addons.settings.AddonSetting;

public class CategorySetting extends AddonSetting {

    public CategorySetting(String text, Addon parent) {
        super(text, parent);


        Shindo.getInstance().getAddonManager().addSettings(this);

    }
}

File: src/main/kotlin/com/shindoclient/shindo/management/settings/impl/CategorySetting.kt
package com.shindoclient.shindo.management.settings.impl

import com.shindoclient.shindo.management.language.TranslateText
import com.shindoclient.shindo.management.settings.Setting
import com.shindoclient.shindo.management.settings.config.ConfigOwner

open class CategorySetting : Setting {
    private var collapsed = false

    constructor(text: TranslateText, parent: ConfigOwner) : super(text, parent)

    constructor(name: String, parent: ConfigOwner) : super(name, parent)

    fun isCollapsed(): Boolean = collapsed

    fun setCollapsed(collapsed: Boolean) {
        this.collapsed = collapsed
    }

    fun toggle() {
        collapsed = !collapsed
    }
}
